storey/flow.py

```diff
@@ -1552,7 +1591,124 @@ def _static_run(*args, **kwargs):
     return _sval._run(*args, **kwargs)
 
 
-class ParallelExecution(Flow):
+class Executor:
+    def __init__(
+        self,
+        max_processes: Optional[int] = None,
+        max_threads: Optional[int] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.runnables: dict[str, ParallelExecutionRunnable] = {}
+        self._runnable_by_name = {}
+
+        self.max_processes = max_processes or os.cpu_count() or 16
+        self.max_threads = max_threads or 32
+        self.num_processes = 0
+        self.num_threads = 0
+
+        self._process_executor_by_runnable_name = {}
+        self._mp_context = None
+
+    def add_runnable(self, runnable: ParallelExecutionRunnable) -> None:
+        if runnable.name not in self.runnables:
+            self.runnables[runnable.name] = runnable
+        else:
+            raise ValueError(f"Model name '{runnable.name}' is not unique")
+
+    def init_runnable(self, runnable: Union[str, ParallelExecutionRunnable]) -> None:
+        if isinstance(runnable, ParallelExecutionRunnable) and runnable.name in self._runnable_by_name:
+            raise ValueError(f"ParallelExecutionRunnable name '{runnable.name}' is not unique")
+        if isinstance(runnable, str) and runnable not in self.runnables:
+            raise ValueError(f"{runnable} is not exists")
+        elif isinstance(runnable, str) and runnable in self._runnable_by_name:
+            return
+        elif isinstance(runnable, str):
+            runnable = self.runnables[runnable]
```

Comment:
I'd refactor it a bit:
```suggestion
        if isinstance(runnable, str):
            if runnable not in self.runnables:
                raise ValueError(f"ParallelExecutionRunnable '{runnable}' does not exist")
            if runnable in self._runnable_by_name:
                return
            runnable = self.runnables[runnable]
        elif runnable.name in self._runnable_by_name:
            raise ValueError(f"ParallelExecutionRunnable name '{runnable.name}' is not unique")
```

Reply:
changed this logic a bit